rlclientlib/live_model_impl.cc

```diff
@@ -145,6 +146,49 @@ namespace reinforcement_learning {
     return error_code::success;
   }
 
+  int live_model_impl::request_slates_decision(const char * context_json, unsigned int flags, slates_response& resp, api_status* status)
+  {
+    const auto uuid = boost::uuids::to_string(boost::uuids::random_generator()());
+    return request_slates_decision(uuid.c_str(), context_json, flags, resp, status);
+  }
+
+  int live_model_impl::request_slates_decision(const char * event_id, const char * context_json, unsigned int flags, slates_response& resp, api_status* status)
+  {
+    if (_learning_mode == IMITATION) {
+      // Imitaion mode is not supported here at this moment
+      return error_code::not_supported;
+    }
+
+    resp.clear();
+    //clear previous errors if any
+    api_status::try_clear(status);
+
+    //check arguments
+    RETURN_IF_FAIL(check_null_or_empty(event_id, status));
+    RETURN_IF_FAIL(check_null_or_empty(context_json, status));
+
+    // Ensure multi comes before slots, this is a current limitation of the parser.
+    RETURN_IF_FAIL(utility::validate_multi_before_slots(context_json, _trace_logger.get(), status));
+
+    size_t num_decisions;
+    RETURN_IF_FAIL(utility::get_slot_count(num_decisions, context_json, _trace_logger.get(), status));
+
+    std::vector<std::vector<uint32_t>> actions_ids;
+    std::vector<std::vector<float>> actions_pdfs;
+    std::string model_version;
+
+    // This will behave correctly both before a model is loaded and after. Prior to a model being loaded it operates in explore only mode.
+    RETURN_IF_FAIL(_model->request_slates_decision(event_id, num_decisions, context_json, actions_ids, actions_pdfs, model_version, status));
+
+    RETURN_IF_FAIL(populate_slates_response(actions_ids, actions_pdfs, std::move(std::string(event_id)), std::move(std::string(model_version)), resp, _trace_logger.get(), status));
```

Comment:
I think you don't need to `std::move` here since `std::string(event_id)` and `std::string(model_version)` are already `rvalue`s

Reply:
Good point, thanks!